### DISCUSSION TITLE
chatterino2-git aur package: please consider toning down the version bump ci job a bit

### DISCUSSION THREAD

**Discussion by Ckath** [Chatterino/chatterino2#2736] [Q&A] [answered]
currently it seems the aur package gets version bumped on **every single git commit** this causes the package to be rebuild constantly for everyone that has the -git installed from the aur.

I did a quick check with some others in #archlinux and everyone agreed vcs packages are really only meant to be updated for packaging changes, not every commit or even really any version change in software for that matter.

my suggestions here would be to test the pkgbuild in ci, but only push to aur unless an error causing it to not build correctly is fixed. not sure how this would be achieved as I'm not that familiar with ci, possibly some special commit tag?

would also be interested what others think as I have gotten some odd stuff before on the `chatterino2-git` package, like it being flagged as 'out-of-date' when there wasnt any packaging issue.

**Comment by leon-richardt** [comment]
I think we should stick to the convention for `*-git` packages

**Reply by pajlada** [reply]
Is there proper documentation for how the `*-git` packages are supposed to behave?

If not, this could be a good wiki topic to add

**Reply by Ckath** [reply]
not specifically as far as I could find, its 'implied' by the pkgver() function always generating the latest version regardless of what the package states. its generally assumed a `-git` will just grab the latest sources
https://wiki.archlinux.org/title/VCS_package_guidelines

**Comment by pajlada** [comment]
We could potentially tone it down to a cronjob instead for now, so push it once per day/once per week if new commits have happened.

The reason we started pushing changes through the CI pipeline was to ensure people actually got the new updates without having to use the `--devel` flag in their AUR wrappers (This was a user complaint).

This could be alleviated by having multiple AUR packages, as other packages normally do.
e.g. `chatterino2`, `chatterino2-nightly-appimage` etc

Would a `chatterino2-nightly-appimage` making updates through the CI fit within the AUR philosophy?

**Reply by Ckath** [reply]
yes, alternative chatterino2 packages that arent `-git` are fine to update all the time `chatterino2-nightly` or `chatterino2` would be fine with more or less the same pkgbuild but version bumped and built from the latest release/nightly sources

**Reply by pajlada** [reply]
But packages that are built through `-git` should always be suffixed by `-git` correct? So renaming the package to `chatterino2-nightly-git` would not "solve" the issue?

**Reply by Ckath** [reply]
the `-git` is meant for packages that are tied to the latest commit generally, plenty (non `-git`) packages on the aur still build from a specific version tag/release.

**Reply by pajlada** [reply]
Right, `-git` packages are tied to the latest commit but the `PKGVER` tag is not _meant_ to be kept up to date (and apparently shouldn't, as per your IRC conversation)

**Reply by pajlada** [reply]
To rephrase, for `-git` packages, the `PKGVER` tag is not meant to be kept up-to-date in the AUR repository (and shouldn't, as per @Ckath's IRC conversation)

**Reply by Ckath** [reply]
the `pkgver()` function on `-git` packages pretty much makes what the `pkgver` var is set to irrelevant as its rechecked each build. hence theres no need to bump it in the pkgbuild

**Reply by pajlada** [reply]
That was where the confusion came from I believe, as users were not receiving 'updates' without us bumping the `pkgver`, rather they had to manually update the package or search for updates using the `--devel` flag.

The solution I had for a long time was manually bumping `pkgver` every now and then, until https://github.com/Chatterino/chatterino2/issues/2491 was implemented.

**Reply by Ckath** [reply]
it would probably be benifitial to make some non `-git` packages, when I first put the `chatterino2-git` package up there werent any fast release cycles yet or many resources. but now it makes more sense to do some regular packages. 

if theres just `chatterino2`/`chatterino2-nightly` which builds from the sources in releases, or some appimage ones like `chatterino2-appimage`/`chatterino2-appimage-nightly`. that would probably be a better option for most users

if you want I could do the pkgbuilds for the former(not that famliar with appimages myself) or if not its just basically a copy of the current `-git` one minus `pkgver()`, with actual md5sums and with a source of `https://github.com/Chatterino/chatterino2/archive/v$pkgver.tar.gz`

**Reply by pajlada** [reply]
Fair enough - we serve a lot of new Linux users (and the package is co-maintained by someone who also isn't too familiar with AUR and its guidelines).

As action items:
1) Disable the pushing from the CI job (can be achieved by setting the GitHub Action aur-sync-action `dry_run` flag to `true`)
2) Create new binary AUR packages
     * `chatterino2-appimage` - Manually revised AUR package, pushed whenever a new stable version of Chatterino2 has been released.
     * `chatterino2-nightly-appimage` - Automatically pushed AUR package, built in CI pushing an updated version of an AppImage file.
3) Pin a comment at the top of the current `chatterino2-git` AUR package, possibly adding some 'echo' postinstall, explaining how VCS packages work

Do you see anything wrong with the above-mentioned action items @micwoj92 ?

Thanks for your understanding and patience as I try to learn.

**Reply by micwoj92** [reply]
Everything seems good, if you need help just @ me. Thanks for sorting it out.

**Reply by Ckath** [reply]
> if you want I could do the pkgbuilds for the former(not that famliar with appimages myself) or if not its just basically a copy of the current `-git` one minus `pkgver()`, with actual md5sums and with a source of `https://github.com/Chatterino/chatterino2/archive/v$pkgver.tar.gz`

turns out the regular idea of packages where you build from a releases src tar gets complicated quite a bit when you throw in submodule build deps. for which the solutions seem to be 'package all the submodules as aur packages' - which isnt great for this, 'build from git anway' - which isnt super clean for 'release' packages, 'do some cryptic stuff to build from the src tarball while adding the submodules form git' - which just seems even worse.

but I think just adding the `-appimage` and `-appimage-nightly` packages, and leaving `-git` updates be unless broken should be all good

**Comment by Ckath** [comment]
would it be possible to put all the submodule libs in the release tar? this would ease the release packages as you can then just build from that directly without jumping through any hoops

**Comment by pajlada** [comment, marked as answer]
`-git` AUR package only bumps on things that break builds now so I'll mark this as fixed.

We also provide `-bin` packages and `-nightly-bin` packages for users who prefer not to build Chatterino themselves.

Thanks again for the PKGBUILD insights!
